src/Microsoft.AspNet.Server.Kestrel/Http/Frame.Generated.cs

```diff
@@ -0,0 +1,297 @@
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Server.Kestrel.Http 
+{
+    public partial class Frame
+    {
+        
+        private object _currentIHttpRequestFeature;
+        private object _currentIHttpResponseFeature;
+        private object _currentIHttpRequestIdentifierFeature;
+        private object _currentIHttpSendFileFeature;
+        private object _currentIServiceProvidersFeature;
+        private object _currentIHttpAuthenticationFeature;
+        private object _currentIHttpRequestLifetimeFeature;
+        private object _currentIQueryFeature;
+        private object _currentIFormFeature;
+        private object _currentIResponseCookiesFeature;
+        private object _currentIItemsFeature;
+        private object _currentIHttpConnectionFeature;
+        private object _currentITlsConnectionFeature;
+        private object _currentIHttpUpgradeFeature;
+        private object _currentIHttpWebSocketFeature;
+        private object _currentISessionFeature;
```

Comment:
Always present:
- IHttpRequestFeature
- IHttpResponseFeature
- IHttpRequestIdentifierFeature
- IServiceProvidersFeature
- IHttpRequestLifetimeFeature (eventually Kestrel will implement this)
- IHttpConnectionFeature (eventually Kestrel will implement this)

Common:
- IHttpAuthenticationFeature
- IQueryFeature
- IFormFeature

Sometimes:
- IResponseCookiesFeature
- IItemsFeature
- ITlsConnectionFeature (Client cert only)
- IHttpUpgradeFeature
- IHttpWebSocketFeature
- ISessionFeature

Rare/Never:
- IHttpSendFileFeature